test/e2e_node/disk_eviction_test.go

```diff
@@ -96,16 +88,26 @@ var _ = framework.KubeDescribe("Kubelet Eviction Manager [Serial] [Disruptive]",
 				})
 			})
 
-			It("should evict the pod using the most disk space [Slow]", func() {
-				if !isImageSupported() {
-					framework.Logf("test skipped because the image is not supported by the test")
+			AfterEach(func() {
+				if !isImageSupported() || !evictionOptionIsSet() { // Skip the after each
 					return
 				}
-				if !evictionOptionIsSet() {
-					framework.Logf("test skipped because eviction option is not set")
-					return
+				podClient.Delete(busyPodName, &api.DeleteOptions{})
```

Comment:
I mean we can rely on the normal cleanup process, instead of the manual cleanup. I don't think longer grace period is required. kubelet should be able to clean up normally, no?